compiler/ml/transl_recmodule.ml

```diff
@@ -66,16 +66,6 @@ let init_shape modl =
                   non_const = cstr_non_const;
                   attrs = [];
                 } )
-        | {desc = Tconstr (p, _, _)} when Path.same p Predef.path_lazy_t ->
```

Comment:
This is what would cause the recursive module test to fail.
I think that's fine, unless the recursive module example pattern is important to support.